generator/internal/descriptor_utils.cc

```diff
@@ -652,15 +652,37 @@ std::string GetEffectiveServiceName(VarsDictionary const& vars,
   return std::string(name.data(), name.size());
 }
 
+/// If a service name mapping exists, return the replacement comment.
+/// If it does not exist, return a null optional.
+/// Parses a command line argument in the form:
+/// {"service_name_to_comments": "service_a=comment_a,service_b=comment_b"}.
+absl::optional<std::string> GetReplacementComment(VarsDictionary const& vars,
+                                                  absl::string_view name) {
+  auto service_name_to_comments = vars.find("service_name_to_comments");
+  if (service_name_to_comments != vars.end()) {
+    std::map<absl::string_view, absl::string_view> map;
+    for (absl::string_view arg :
+         absl::StrSplit(service_name_to_comments->second, ',')) {
+      map.insert(absl::StrSplit(arg, absl::MaxSplits('=', 1)));
+    }
+    auto result = map.find(name);
+    if (result != map.end()) {
+      return std::string(result->second.data(), result->second.size());
+    }
+  }
```

Comment:
This is a $O(n ln(n) + ln(n))$ algorithm when you can do a $O(n)$

```suggestion
    for (absl::string_view arg :
         absl::StrSplit(service_name_to_comments->second, ',')) {
      std::pair<absl::string_view, absl::string_view> p = absl::StrSplit(arg, absl::MaxSplits('=', 1));
      if (p.first == name) return std::string(p.second.data(), p.second.size());
    }
    return absl::nullopt;
```

Reply:
DOne.

Reply:
Did for above as well.